core/controllers/reader.py

```diff
@@ -473,6 +473,20 @@ class StateHitEventHandler(base.BaseHandler):
 
     REQUIRE_PAYLOAD_CSRF_CHECK = False
 
+    URL_PATH_ARGS_SCHEMAS = {
+        'username': {
+            'schema': {
+                'type': 'basestring',
+                'validators': [{
+                    'id': 'is_valid_username_string'
+                }]
+            }
+        }
```

Comment:
Why username? When the arg of the `post` function is `exploration_id`?

Reply:
done